File: test/fixtures/stdlib/sys_extended.py
from pylib import sys
from pylib.assertions import py_assert_stdout


def main() -> None:
    old_argv = sys.argv
    old_path = sys.path
    sys.set_argv(["a", "b"])
    sys.set_path(["x"])
    print(isinstance(sys.argv, list))
    print(isinstance(sys.path, list))
    print(sys.argv[0])
    print(sys.path[0])
    sys.set_argv(old_argv)
    sys.set_path(old_path)


def _case_main() -> None:
    main()


if __name__ == "__main__":
    print(py_assert_stdout(["True", "True", "a", "x"], _case_main))

File: test/fixtures/stdlib/sys_extended.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Any = System.Object;
using int64 = System.Int64;
using float64 = System.Double;
using str = System.String;
using Pytra.CsModule;
using sys = Pytra.CsModule.sys;

public static class Program
{
    public static bool run_sys_extended()
    {
        System.Collections.Generic.List<bool> checks = new System.Collections.Generic.List<bool>();
        var old_argv = args;
        var old_path = sys.path;
        sys.set_argv(new System.Collections.Generic.List<string> { "a", "b" });
        sys.set_path(new System.Collections.Generic.List<string> { "x" });
        checks.Add(System.Object.Equals(Pytra.CsModule.py_runtime.py_runtime_value_isinstance(args, Pytra.CsModule.py_runtime.PYTRA_TID_LIST), true));
        checks.Add(System.Object.Equals(Pytra.CsModule.py_runtime.py_runtime_value_isinstance(sys.path, Pytra.CsModule.py_runtime.PYTRA_TID_LIST), true));
        checks.Add(System.Object.Equals(args[System.Convert.ToInt32(0)], "a"));
        checks.Add(System.Object.Equals(sys.path[System.Convert.ToInt32(0)], "x"));
        sys.set_argv(old_argv);
        sys.set_path(old_path);
        return System.Linq.Enumerable.All(checks, __x => System.Convert.ToBoolean(__x));
    }
    
    public static void Main(string[] args)
    {
            System.Console.WriteLine(run_sys_extended());
    }
}
